oolab/src/main/java/agh/ics/oop/SimulationEngine.java

```diff
@@ -10,7 +10,11 @@ public SimulationEngine(MoveDirection[] moveDirection, IWorldMap map, Vector2d[]
         this.moveDirection = moveDirection;
         this.positions = positions;
         for (Vector2d v : positions){
-            map.place(new Animal(map, v));
+            Animal a = new Animal(map, v);
+            map.place(a);
+            System.out.println(a.getPosition());
+            a.addObserver((IPositionChangeObserver) map);
```

Comment:
Jeśli każda mapa rozszerzałaby ten interfejs to nie musielibyśmy rzutować -> interfejsy też mogą rozszerzać inne interfejsy.